ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java

```diff
@@ -890,15 +890,14 @@ public CompletableFuture<RaftClientReply> submitClientRequestAsync(
     }
 
     final RaftClientRequest.Type type = request.getType();
-    replyFuture.whenComplete((clientReply, exception) -> {
+    return replyFuture.whenComplete((clientReply, exception) -> {
       if (clientReply.isSuccess()) {
```

Comment:
BTW, it should stop the timer even if it has failed.  Could you also change it?
```java
-      if (clientReply.isSuccess()) {
-        timerContext.ifPresent(Timekeeper.Context::stop);
-      }
+      timerContext.ifPresent(Timekeeper.Context::stop);
```

Reply:
Thanks for reviewing, updated.